src/plugins/workspace/public/components/workspace_form/workspace_permission_setting_input.tsx

```diff
@@ -118,23 +135,52 @@ export const WorkspacePermissionSettingInput = ({
   );
 
   const handlePermissionModeOptionChange = useCallback(
-    (id: string) => {
-      if (optionIdToWorkspacePermissionModesMap[id]) {
-        onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+    (option: Array<EuiComboBoxOptionOption<any>>) => {
+      if (option.length > 0) {
+        const id = option[0].value;
+        if (optionIdToWorkspacePermissionModesMap[id]) {
+          onPermissionModesChange([...optionIdToWorkspacePermissionModesMap[id]], index);
+        }
       }
     },
     [index, onPermissionModesChange]
   );
 
+  const toggleTypeList = () => {
+    if (!isTypeListOpen) {
+      if (type === WorkspacePermissionItemType.User) {
+        setTypeOptions([{ ...typeOptions[0], checked: 'on' }, typeOptions[1]]);
```

Comment:
Instead of marking the option with `checked: 'on'`,  the `props.type` passed to this component already stores the selections state, I think we can rely on that.

Reply:
Never mind, it needs to be marked with `checked`

Reply:
But for the first time to dropdown the list, nothing is selected, maybe we could use useEffect?

Reply:
Can we use `useMemo` to calculate the typeOptions? I think we don't need a typeOptions state. The `checked` property can follow the `props.type`.

Reply:
thanks, it's a better approach!